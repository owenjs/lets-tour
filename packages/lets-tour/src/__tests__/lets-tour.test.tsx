import React, { FC } from "react";
import { fireEvent, render, screen, waitFor } from "@testing-library/react";
import { ILetsTourProviderProps, LetsTourProvider, TLetsTourStep, useLetsTourContext } from "../index";

window.HTMLElement.prototype.scrollIntoView = jest.fn();

describe("Lets Tour", () => {
  let STEPS: TLetsTourStep[];
  let Popover: FC;

  const exec = (props?: Omit<ILetsTourProviderProps, "steps" | "Component">) => {
    const App = () => {
      const { handleStartTour } = useLetsTourContext();

      return (
        <>
          <div data-tour-step="1">Step 1 Item</div>
          <div data-tour-step="2">Step 2 Item</div>

          <button onClick={handleStartTour}>Open Tour</button>
        </>
      );
    };

    return render(
      <LetsTourProvider steps={STEPS} Component={Popover} {...props}>
        <App />
      </LetsTourProvider>,
      {
        container: document.body
      }
    );
  };

  beforeEach(() => {
    STEPS = [
      {
        selector: `[data-tour-step="1"]`
      },
      {
        selector: `[data-tour-step="2"]`
      }
    ];

    Popover = () => <div>Popover</div>;
  });

  it("should render without popover", () => {
    const { container } = exec();

    expect(screen.queryByText("Popover")).toBeNull();

    expect(container).toMatchSnapshot();
  });

  it("should render popover if open initially", () => {
    const { container } = exec({
      isOpen: true
    });

    waitFor(() => {
      expect(screen.queryByText("Popover")).not.toBeNull();
    });

    expect(container).toMatchSnapshot();
  });

  it("should render popover if user triggers 'handleStartTour'", () => {
    exec();

    fireEvent.click(screen.getByText("Open Tour"));

    expect(screen.queryByText("Popover")).not.toBeNull();
  });
});
