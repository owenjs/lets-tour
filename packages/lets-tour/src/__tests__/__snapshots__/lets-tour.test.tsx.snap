// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lets Tour should render popover if open initially 1`] = `
<body>
  <div
    style="position: absolute; left: 0px; top: 0px; z-index: 100000;"
  >
    <div>
      Popover
    </div>
  </div>
  <div
    style="opacity: 0.7; left: 0px; top: 0px; position: fixed; z-index: 99999; pointer-events: none; color: rgb(0, 0, 0);"
  >
    <svg
      height="768"
      style="width: 1024px; height: 768px; left: 0px; top: 0px; position: fixed;"
      width="1024"
      xmlns="http://www.w3.org/2000/svg"
    >
      <defs>
        <mask
          id=":r0:"
        >
          <rect
            fill="white"
            height="768"
            width="1024"
            x="0"
            y="0"
          />
          <rect
            style="width: 20px; height: 20px; fill: black;"
          />
        </mask>
        <clippath
          id=":r1:"
        >
          <polygon
            points="0 0, 0 768, 0 768, 0 0, 20 0, 20 20, 0 20, 0 768, 1024 768, 1024 0"
          />
        </clippath>
      </defs>
      <rect
        style="width: 1024px; height: 768px; fill: currentColor; mask: url(#:r0:);"
      />
      <rect
        style="width: 1024px; height: 768px; fill: currentcolor; pointer-events: auto; clip-path: url(#:r1:);"
      />
      <rect
        style="width: 20px; height: 20px; pointer-events: auto; fill: transparent; display: none;"
      />
    </svg>
  </div>
  <div
    data-tour-step="1"
  >
    Step 1 Item
  </div>
  <div
    data-tour-step="2"
  >
    Step 2 Item
  </div>
  <button>
    Open Tour
  </button>
</body>
`;

exports[`Lets Tour should render without popover 1`] = `
<body>
  <div
    data-tour-step="1"
  >
    Step 1 Item
  </div>
  <div
    data-tour-step="2"
  >
    Step 2 Item
  </div>
  <button>
    Open Tour
  </button>
</body>
`;
